easybuild/easyblocks/m/matlab.py

```diff
@@ -141,24 +141,56 @@ def install_step(self):
             keys = keys.split(',')
 
         # Make one install for each key
-        for key in keys:
-            cmd = ' '.join([
-                self.cfg['preinstallopts'],
-                src,
-                '-v',
-                tmpdir,
-                '-inputFile',
-                self.configfile,
-                '-fileInstallationKey',
-                key,
-                self.cfg['installopts'],
-            ])
-            (out, _) = run_cmd(cmd, log_all=True, simple=False)
+        for i, key in enumerate(keys):
+
+            self.log.debug('Installing with key %s of %s', i + 1, len(keys))
+
+            if LooseVersion(self.version) >= LooseVersion('2020a'):
+                self.log.debug('Version is %s - using binary installer method', self.version)
+                try:
+                    with tempfile.NamedTemporaryFile() as fd:
+                        tmp_configfile = fd.name
+                        with open(self.configfile) as template_fd:
+                            tmp_config = template_fd.read()
+                        tmp_config = tmp_config.replace('# fileInstallationKey=', 'fileInstallationKey=%s' % key)
```

Comment:
We had an internal discussion about whether to use a string replacement here, or maintain uniformity with the config section earlier in the block and use regex substitution.

Reply:
A regexp is much safer to use. Please change to that.